### PR TITLE
feat: Add enhancements to Legend

### LINKED ISSUE #3697: ChartPanel needs ability to display legend at bottom

### LINKED ISSUE #3710: ChartPanel scrollbar alignment issue
image

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #3710
Fixes #3697

Adds 6 symbol legend shapes
Adds label alignment options
Adds bottom legend feature to ChartPanel

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [ ] The commit is signed.
- [ ] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by GordonSmith** [comment]
Some quick observations:
1.  Default Icon Size (Circle) is too large + Total looks misplaced:
image
2.  Switching between placement is broken:
image
image

File: packages/layout/src/ChartPanel.ts [hunk 1]
```diff
@@ -554,6 +587,7 @@ ChartPanel.prototype.publishProxy("progress_blurBar", "_progressBar", "blurBar")
 ChartPanel.prototype.publishProxy("progress_blurSize", "_progressBar", "blurSize");
 ChartPanel.prototype.publishProxy("progress_blurColor", "_progressBar", "blurColor");
 ChartPanel.prototype.publishProxy("progress_blurOpacity", "_progressBar", "blurOpacity");
+ChartPanel.prototype.publish("legendPosition", "right", "set", "Position of legend (must be set before initial render)", ["right", "bottom"]);
```
Comment (GordonSmith) [inline packages/layout/src/ChartPanel.ts:590]: Move PP so its with the other legend items
Rename so its consistent with the other legend items

File: packages/layout/src/Border2.css [hunk 2]
```diff
@@ -6,22 +10,17 @@
 }
 
 .layout_Border2 > .body > .center {
-    flex: 3 1 60%;
+    flex: 1;
     -ms-flex: 3 1 auto;
```
Comment (GordonSmith) [inline packages/layout/src/Border2.css:14]: should the ms specific be changed?

File: packages/layout/src/Border2.css [hunk 3]
```diff
@@ -6,22 +10,17 @@
...
-    order: 2;
 }
 
 .layout_Border2 > .body > .lhs {
-    flex: 1 6 20%;
     -ms-flex: 1 6 auto;
```
Comment (GordonSmith) [inline packages/layout/src/Border2.css:18]: should the ms specific be removed?

File: packages/layout/src/Border2.css [hunk 4]
```diff
@@ -6,22 +10,17 @@
...
-    order: 1;
 }
 
 .layout_Border2 > .body > .rhs {
     margin-left: 4px;
-    flex: 1 6 20%;
     -ms-flex: 1 6 auto;
```
Comment (GordonSmith) [inline packages/layout/src/Border2.css:23]: should the ms specific be removed?

File: packages/layout/src/Border2.ts [hunk 5]
```diff
@@ -2,9 +2,13 @@ import { BBox, HTMLWidget, Platform, Widget } from "@hpcc-js/common";
 
 import "../src/Border2.css";
 
+export type IOverflow = "hidden" | "scroll" | "visible" | "auto";
```
Comment (GordonSmith) [inline packages/layout/src/Border2.ts:5]: Rename to OverflowT for consistency

File: packages/layout/src/Border2.ts [hunk 6]
```diff
@@ -2,9 +2,13 @@ import { BBox, HTMLWidget, Platform, Widget } from "@hpcc-js/common";
...
+export type IChartPanelSection = "top" | "right" | "bottom" | "left" | "center";
```
Comment (GordonSmith) [inline packages/layout/src/Border2.ts:6]: Rename to ChartPanelSectionT for consistency

**Comment by GordonSmith** [comment]
@jbrundage can you add a few extra examples to the Gallery for these?

File: packages/layout/src/Border2.css (hunk 2 above)
Comment (jbrundage) [reply packages/layout/src/Border2.css:14]: If IE11 is our target IE browser then it should actually be removed.

Comment (GordonSmith) [reply packages/layout/src/Border2.css:14]: Yup - I do know that the defaults in IE are diff to chrome:
IE:  "0 0 auto" 
Chrome:  "0 1 auto"

So you may need something?

**Comment by jbrundage** [comment]
@GordonSmith added 5 gallery demos (which lead to fixing a few alignment issues). It's ready for review

**Comment by GordonSmith** [comment]
@jbrundage those gallery items are missing?

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**